spec/lib/orderly_spec.rb

```diff
@@ -33,6 +33,18 @@
       expect { expect(this).to appear_before(that) }.to raise_error error_text
       expect { expect(this).not_to appear_before(that) }.to raise_error error_text
     end
-  end
 
+    context 'handling within blocks' do
+
+      specify "using a Capybara within block" do
+        page.visit "/thisthatthis"
+        expect(that).to appear_before this, within: '.within-block'
+      end
+
+      specify "not using a Capybara within block" do
+        page.visit "/thisthatthis"
+        expect(that).not_to appear_before this
```

Comment:
This needs the `:within` option and should have the order of `this` and `that` swapped in order to test scoping. This passes because `String#index` stops at the first occurrence of `this` and doesn't ensure that the position of `that` inside the `<div>` is higher than `this` inside the `<div>`, even though there is one outside the `<div>`.